### ISSUE TITLE
Number of specified iterations (10000) exceeds the maximum possible number of iterations (6)

### ISSUE THREAD

**Issue by aeladly91** [0todd0000/spm1d#187]
Hello,
I am analyzing single channel ECoG data obtained from a healthy and a diseased rat's cortices. I obtained the spectrogram of the recordings from the healthy and diseased rat. And I used your spm1d package to perform statistical inference on the those spectrograms to see if they are statistically different from each other. I followed your spm1d tutorial, the code worked fine, but it started complaining when it gotten to this line of code.
 snpmi   = snpm.inference(0.05, two_tailed=True, iteration = 100000 )

it returned this error 

Number of specified iterations (10000) exceeds the maximum possible number of iterations (6)

My spectrograms dataset dimensions are 8 X 600X 3800 (each spectrogram is a 600X3800 matrix) and there are 8 of them 4 for the healthy and 4 for the diseased rat.

Can you suggest what I am doing wrong?

Thanks in advance

Ahmed

**Comment by 0todd0000** [comment]
Sorry for the delay!

The error means that there are too few observations to run 100,000 unique iterations, and that there are only 6 unique iterations. It sounds like you are conducting a paired test, which means that there are actually just 4 observations (i.e., one difference observation for each rat). If the healthy and diseased rats can indeed be paired, then a paired test is appropriate, but if they cannot be appropriately paired, then you could try a two-sample test instead. This will raise the number of observations from 4 to 8.

For the two-sample case, the total number of unique iterations is:

<img width="73" alt="eq" src="https://user-images.githubusercontent.com/4276626/131951417-f9619870-d0c0-4387-b5ec-cc04316f9dc2.png">

where *N* is the number of unique iterations, and *n1* and *n2* are the sample sizes of the two groups. If *n1*=*n2*=4, then *N*=70, so the maximum value you could set is `iterations=70`.

In other words, the sample size is rather small. Usually sample sizes of at least *n1*=*n2*=8 or more are needed, so that $N$ exceeds 10,000. For *n1*=*n2*=8, the number of unique iterations is *N*=12,870. Using *n1*=*n2*=8 is just a rule-of-thumb for achieving relatively stable numerical results.

**Comment by aeladly91** [comment]
Dear Todd,

Thank you so much for your reply. Yes that explains it. Thank you so much for the spm1d package and making it open to everyone to use.

Have a great day.

Ahmed

**Comment by 0todd0000** [comment]
You're welcome!  Thanks for the feedback.
